src/SIM/SIMbase.C

```diff
@@ -1579,8 +1579,9 @@ void SIMbase::getBoundaryNodes (int pcode, IntVec& glbNodes, Vec3Vec* XYZ) const
   ASMbase* pch;
   size_t i, node, last = 0;
   for (PropertyVec::const_iterator p = myProps.begin(); p != myProps.end(); ++p)
-    if (abs(p->pindx) == pcode && (pch = this->getPatch(p->patch)))
-      switch (pch->getNoParamDim() - abs(p->ldim)) {
+    if (abs(p->pindx) == pcode && (pch = this->getPatch(p->patch))) {
+      char dim = pch->getNoParamDim() - abs(p->ldim);
+      switch (dim) {
```

Comment:
Is it the unsigned vs. signed problem again? Then better with:

switch((char)pch->getNoParamDim() - abs(p->ldim))

such that is clear (and only one line changed instead of 4;-)